modules/seqexec/server/src/main/scala/seqexec/server/gsaoi/Gsaoi.scala

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2016-2019 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package seqexec.server.gsaoi
+
+import cats.data.Reader
+import cats.effect.Sync
+import cats.effect.LiftIO
+import cats.implicits._
+import edu.gemini.spModel.config2.Config
+import edu.gemini.spModel.gemini.gsaoi.Gsaoi._
+import edu.gemini.spModel.obscomp.InstConstants.DARK_OBSERVE_TYPE
+import edu.gemini.spModel.obscomp.InstConstants.OBSERVE_TYPE_PROP
+import edu.gemini.spModel.seqcomp.SeqConfigNames.OBSERVE_KEY
+import gem.enum.LightSinkName
+import java.lang.{Double => JDouble}
+import java.lang.{Integer => JInt}
+import shapeless.tag
+import seqexec.server.ConfigUtilOps._
+import seqexec.model.dhs.ImageFileId
+import seqexec.model.enum.Instrument
+import seqexec.server.ConfigUtilOps.ExtractFailure
+import seqexec.server.ConfigResult
+import seqexec.server.InstrumentSystem
+import seqexec.server.ObserveCommand
+import seqexec.server.Progress
+import seqexec.server.SeqActionF
+import seqexec.server.SeqObserveF
+import seqexec.server.TrySeq
+import seqexec.server.keywords.DhsClient
+import seqexec.server.keywords.DhsInstrument
+import seqexec.server.keywords.KeywordsClient
+import seqexec.server.InstrumentSystem._
+import seqexec.server.gsaoi.GsaoiController._
+import seqexec.server.tcs.FOCAL_PLANE_SCALE
+import squants.space.Arcseconds
+import squants.{Length, Time}
+import squants.time.TimeConversions._
+
+final case class Gsaoi[F[_]: LiftIO: Sync](
+  controller: GsaoiController[F],
+  dhsClient:  DhsClient[F])
+    extends DhsInstrument[F]
+    with InstrumentSystem[F] {
+
+  import Gsaoi._
+
+  override def sfName(config: Config): LightSinkName = LightSinkName.Gsaoi
+
+  override val contributorName: String = "GSAOI"
+
+  override val observeControl: InstrumentSystem.ObserveControl[F] =
+    OpticControl[F](StopObserveCmd[F](SeqActionF.liftF(controller.stopObserve)),
+                 AbortObserveCmd[F](SeqActionF.liftF(controller.abortObserve)),
+                 PauseObserveCmd[F](SeqActionF.void),
+                 ContinuePausedCmd[F](_ => SeqActionF(ObserveCommand.Unsupported)),
```

Comment:
GSAOI doesn't support pause even being optical, etc. so I'm returning unsupported though it should never be called. We could instead do another trait like `PartialOpticControl`, I don't know if it would be simpler